vmdb/app/controllers/ops_controller/diagnostics.rb

```diff
@@ -133,6 +133,8 @@ def log_depot_edit
         replace_right_cell(x_node)
       end
     when "validate"
+      id = params[:id] ? params[:id] : "new"
+      return unless load_edit("logdepot_edit__#{id}", "replace_cell__explorer")
```

Comment:
@h-kataria Wow, I'd never know to do this.  Thanks.  

I'm not sure if I should open github issues since I'm not familiar with this code but these things jump out at me since I'm pretty new to what this code is doing.

1) It would be nice to normalize these values:

We're in the method: `log_depot_edit`
We do `assert_privileges("...log_depot_edit")` at the top of the method
At the bottom of this method, we're doing `replace_right_cell("log_depot_edit")`
We're doing `logdepot_edit__` here (without an underscore)

If they matched, we could use `__method__` in the 3 places.

2) Also, I'd love to see `load_edit` return the contents of `@edit` since we use `@edit` right after to pull out specific settings.  I've been trying to convert side effect methods to query methods (methods that return data instead of modifying state).  Side effect methods make code very complicated.

3) Additionally, it would simplify this code by making `load_edit` and related methods elsewhere build this `"logdepot_edit__#{id}"` string via a method so it can handle defaulting to `new` if a value for the id is not provided.

We do this string building of "something_edit_#{somethod.id || "new"}" a lot (perhaps 68 times)... I'd love to understand what's going on here to help simplify this code:
```ruby
# git grep -E "_edit.+\|\|.+new"
...
app/controllers/miq_ae_customization_controller/dialogs.rb:    @edit[:key] = "dialog_edit__#{@record.id || "new"}"
app/controllers/miq_ae_customization_controller/dialogs.rb:    @edit[:key] = "dialog_edit__#{@record.id || "new"}"
app/controllers/miq_ae_customization_controller/old_dialogs.rb:    @edit[:key] = "dialog_edit__#{@dialog.id || "new"}"
app/controllers/miq_policy_controller/alert_profiles.rb:    @edit[:key] = "alert_profile_edit__#{@alert_profile.id || "new"}"
app/controllers/miq_policy_controller/alerts.rb:    @edit[:key] = "alert_edit__#{@alert.id || "new"}"
app/controllers/miq_policy_controller/conditions.rb:    @edit[:key] = "condition_edit__#{@condition.id || "new"}"
app/controllers/miq_policy_controller/events.rb:    @edit[:key] = "event_edit__#{@event.id || "new"}"
app/controllers/miq_policy_controller/miq_actions.rb:    @edit[:key] = "action_edit__#{@action.id || "new"}"
app/controllers/miq_policy_controller/policies.rb:    @edit[:key] = "policy_edit__#{@policy.id || "new"}"
app/controllers/miq_policy_controller/policy_profiles.rb:    @edit[:key] = "profile_edit__#{@profile.id || "new"}"
app/controllers/miq_proxy_controller.rb:    @edit[:key] = "proxy_edit__#{@miq_proxy.id || "new"}"
app/controllers/ops_controller.rb:        redirect_to(:action=>"ap_edit", :tab=>params[:tab], :id=>"#{@scan.id || "new"}")
app/controllers/ops_controller.rb:        redirect_to(:action=>"ap_edit", :tab=>"edit#{params[:tab]}", :id=>"#{@scan.id || "new"}")
...
```